### PR TITLE
feat(fixtures): fixtures data for customers

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Issue - https://github.com/agritheory/test_utils/issues/10
Collected fixtures data for customers from Autoreader, beam, inventory_tools, electronic_payments

### CODE REVIEW

**Comment by MyuddinKhatri** [comment]
@agritheory for tax_id under customer i have directly stores in fixtures instead of generating it randomly. Please let me know if I should keep this as it is change it to generate randomly.

**Comment by MyuddinKhatri** [comment]
@agritheory should we add separate create_address() and create_contact() as both suppliers and customers have addresses?We can call create_suppliers, create_customers first then create_address and create_contact ?

**Comment by Alchez** [comment]
@MyuddinKhatri I think it's better to just have the `create_suppliers` and `create_customers` commands create the downstream documents, since it seems less likely that an app would need just addresses or contacts.

File: test_utils/utils/setup_fixtures.py [hunk 1]
```diff
@@ -6,12 +6,72 @@
...
+					address_link.get("link_doctype") == "Customer"
+					and address_link.get("link_name") == cust.customer_name
+				):
+					addr = frappe.new_doc("Address")
+					addr.update(address)
+					addr.save()
+
+		for user in users:
+			if frappe.db.exists("User", user.get("username")):
+				continue
```
Comment (Alchez) [inline test_utils/utils/setup_fixtures.py:52]: If the user already exists and the JSON roles are updated to include `Customer`, should this also append that role before returning?

File: test_utils/utils/setup_fixtures.py [hunk 2]
```diff
@@ -6,12 +6,72 @@
...
+					user_doc = frappe.new_doc("User")
+					user_doc.update(user)
+					user_doc.save()
+
+		for contact in contacts:
+			existing_contact = frappe.db.get_value(
+				"Contact", {"email_id": customer.get("email")}
+			)
+			if existing_contact:
+				continue
```
Comment (Alchez) [inline test_utils/utils/setup_fixtures.py:64]: Similar to above. If links are added for a customer, should this also append a row for the link?

File: test_utils/utils/setup_fixtures.py [hunk 3]
```diff
@@ -6,12 +6,72 @@
...
+		cust = frappe.new_doc("Customer")
+		cust.update(customer)
+		cust.save()
+
+		for address in addresses:
+			existing_address = frappe.get_value(
+				"Address", {"address_line1": address.get("address_line1")}
+			)
+			if existing_address:
+				continue
```
Comment (Alchez) [inline test_utils/utils/setup_fixtures.py:39]: Same as contact.

**Review by Alchez — CHANGES_REQUESTED** [review]

**Review by Alchez — APPROVED** [review]

**Comment by agritheory** [comment]
@MyuddinKhatri Can you rebase this to fix the pre-commit spelling issue?

**Comment by MyuddinKhatri** [comment]
> @MyuddinKhatri Can you rebase this to fix the pre-commit spelling issue?

@agritheory codespell is complaining about `pheonix` Is this intentional?

**Comment by agritheory** [comment]
> @agritheory codespell is complaining about pheonix Is this intentional?

No, let's change it to the most standard spelling. The number and variation of spellings of this particular word is wild. Here's an example that's local to me: 

image

**Pull request merged**